scripts/dts/edtlib.py

```diff
@@ -1144,6 +1224,107 @@ def _check_undeclared_props(self):
                      .format(prop_name, self._node.path, self.edt.dts_path,
                              self.binding_path))
 
+    def _init_gpio_ranges(self):
+        # Initialize self.pinctrl_gpio_ranges
+
+        if not hasattr(self, 'pinctrl_gpio_ranges'):
+            self.pinctrl_gpio_ranges = []
+
+        if 'gpio-ranges' not in self.props:
+            return
+
+        for prop_value in self.props['gpio-ranges'].val:
+            entry = ControllerAndData()
+            entry.node = prop_value.node
+            entry.name = prop_value.name
+            entry.controller = self
+            entry.data = prop_value.data
+            # Add to pin controller
+            if not hasattr(prop_value.controller, 'pinctrl_gpio_ranges'):
+                prop_value.controller.pinctrl_gpio_ranges = []
+            prop_value.controller.pinctrl_gpio_ranges.append(entry)
+
+    def _init_clock_outputs(self):
+       # Initialize self.clock_outputs
+
+        self.clock_outputs = []
+
+        node = self._node
+        if "#clock-cells" not in node.props:
+            return
+
+        clock_cells = node.props["#clock-cells"].to_num()
+        clock_output_names = node.props.get("clock-output-names", [])
+        if clock_output_names:
+            clock_output_names = clock_output_names.to_strings()
```

Comment:
Could make it a bit easier to read like this:

```python
if "clock-output-names" in node.props:
    clock_output_names = node.props["clock-output-names"].to_strings()
else:
    clock_output_names = []
```

I try to avoid the "set a variable to a value that might be wrong, do a check, and then overwrite it with the right value" pattern. Gets confusing when scanning for where a variable is set.

Reply:
Changed programming style to be in line.